pkg/kf/testutil/integration.go

```diff
@@ -93,6 +103,123 @@ func RunIntegrationTest(t *testing.T, test func(ctx context.Context, t *testing.
 	test(ctx, t)
 }
 
+// RunKubeAPITest is for executing tests against the Kubernetes API directly.
+func RunKubeAPITest(t *testing.T, test func(ctx context.Context, t *testing.T)) {
+	if !strings.HasPrefix(t.Name(), "TestIntegration_") {
+		// We want to enforce a convention so scripts can single out
+		// integration tests.
+		t.Fatalf("Integration tests must have the name format of 'TestIntegration_XXX`")
+	}
+
+	t.Helper()
+	if testing.Short() {
+		t.Skip()
+	}
+
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	clientCfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+	restCfg, err := clientCfg.ClientConfig()
+	if err != nil {
+		t.Fatalf("couldn't fetch kubeconfig: %v", err)
+	}
+
+	start := time.Now()
+	defer func() {
```

Comment:
nit: Now that this is in multiple places, can we break it out to a helper method?